browser/ui/webui/brave_education/getting_started_helper.cc

```diff
@@ -0,0 +1,136 @@
+
+constexpr auto kTrafficAnnotation = net::DefineNetworkTrafficAnnotation(
+    "brave_education_getting_started_helper",
+    R"(
+      semantics {
+        sender: "Brave Education"
+        description: "Attempts to fetch the content for the Brave Education
+          getting-started page to ensure that it loads successfully."
+        trigger:
+          "Completing the Brave Welcome UX flow."
+        data:
+          "No data sent, other than URL of the getting-started page. "
+          "Data does not contain PII."
+        destination: BRAVE_OWNED_SERVICE
+      }
+      policy {
+        cookies_allowed: NO
+        setting:
+          "None"
+      }
+    )");
+
+std::optional<EducationContentType> GetEducationContentType() {
+  if (base::FeatureList::IsEnabled(features::kShowGettingStartedPage)) {
+    return EducationContentType::kGettingStarted;
+  }
+  return std::nullopt;
+}
+
+}  // namespace
+
+GettingStartedHelper::GettingStartedHelper(Profile* profile)
+    : profile_(profile) {}
+
+GettingStartedHelper::~GettingStartedHelper() = default;
+
+void GettingStartedHelper::GetEducationURL(GetEducationURLCallback callback) {
+  // Add the callback to our list of pending callbacks.
+  url_callbacks_.push_back(std::move(callback));
+
+  // If we are currently waiting on a URL, then exit. All callbacks in the
+  // pending list will be executed when the website check is completed.
+  if (url_loader_) {
+    return;
+  }
+
+  auto content_type = GetEducationContentType();
+  if (!content_type) {
+    RunCallbacks(std::nullopt);
+    return;
+  }
+
+  // Attempt to fetch the content URL in the background.
+  auto url_loader_factory = profile_->GetURLLoaderFactory();
+  auto request = std::make_unique<network::ResourceRequest>();
+  request->url = GetEducationContentServerURL(*content_type);
+  request->referrer_policy = net::ReferrerPolicy::NO_REFERRER;
+  request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+
+  if (auto* delegate = profile_->GetReduceAcceptLanguageControllerDelegate()) {
+    auto languages = delegate->GetUserAcceptLanguages();
+    if (!languages.empty()) {
+      request->headers.SetHeader(request->headers.kAcceptLanguage,
+                                 languages.front());
+    }
+  }
+
+  url_loader_ =
+      network::SimpleURLLoader::Create(std::move(request), kTrafficAnnotation);
+
+  url_loader_->DownloadToString(
+      url_loader_factory.get(),
+      base::BindOnce(&GettingStartedHelper::OnURLResponse,
+                     base::Unretained(this), *content_type),
+      kMaxDownloadBytes);
+}
+
+void GettingStartedHelper::OnURLResponse(EducationContentType content_type,
+                                         std::optional<std::string> body) {
+  if (URLLoadedWithSuccess() && body) {
+    RunCallbacks(GetEducationContentBrowserURL(content_type));
+  } else {
+    RunCallbacks(std::nullopt);
+  }
+}
+
+bool GettingStartedHelper::URLLoadedWithSuccess() {
+  if (!url_loader_) {
```

Comment:
I feel this should just be a `CHECK(url_loader_)`.

Reply:
Sounds good. 👍